### PR TITLE
feat: xcm refactoring

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* refactor: refactoring the way to send XCM transactions and fetch XCM token balances to the new architecture
* fix: removed legacy XCM related files
* fix: modified direct references issues that discussed [here](https://github.com/AstarNetwork/astar-apps/pull/482#discussion_r944398319)
* fix: updated EVM network configuration data to avoid getting warning in MetaMask ([commit](https://github.com/AstarNetwork/astar-apps/pull/510/commits/3578255575782c68a3e3f22070853c0915cc454e))
  * ref: https://gyazo.com/d4db5737b7f0a3b3667c77c4b9af834c 

**Check list**
- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

### CODE REVIEW

File: src/modules/xcm/tokens/index.ts [hunk 1]
```diff
@@ -95,7 +95,8 @@ export const xcmToken = {
       logo: 'https://assets.coingecko.com/coins/images/17172/small/karura.jpeg?1626782066',
       isXcmCompatible: true,
       originChain: 'Karura',
-      minBridgeAmount: '0.1',
+      // ED: 0.1 KAR (Users can't withdraw 0.1KAR to 0 KAR account due to a fee)
+      minBridgeAmount: '0.11',
```
Comment (hoonsubin) [inline src/modules/xcm/tokens/index.ts:99]: is there really no way to get this dynamically from the chain? Is it part of the XCM config?

File: src/v2/services/implementations/XcmEvmService.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import { Guard } from 'src/v2/common';
+import { BusyMessage, ExtrinsicStatusMessage, IEventAggregator } from 'src/v2/messaging';
+import { IGasPriceProvider, IXcmEvmService, TransferParam } from 'src/v2/services';
+import { Symbols } from 'src/v2/symbols';
+import Web3 from 'web3';
+import { TransactionConfig } from 'web3-eth';
+import { AbiItem } from 'web3-utils';
+
+// XCM precompiled contract address
+const PRECOMPILED_ADDR = '0x0000000000000000000000000000000000005004';
```
[the remark this answers is not in the document]
Comment (hoonsubin) [reply src/v2/services/implementations/XcmEvmService.ts:21]: @impelcrypto or is this the precompiled from Moonbeam and not us?

[the remark this answers is not in the document]
Comment (impelcrypto) [reply src/v2/services/implementations/XcmEvmService.ts:21]: It's our precompiled address. We should update it(0x...5004) on our docs as well. https://docs.astar.network/docs/EVM/precompiles/#dappsstaking-precompile

**Review by hoonsubin — APPROVED** [review]

**Review by bobo-k2 — APPROVED** [review]
LGTM

**Review by sirius651 — APPROVED** [review]
LGTM

**Pull request merged**